wordpress-mobile/WordPress-Android
# My Site Dashboard: Phase 2: Metrics - Track Post Card Footer Link Tapped

Fixes #15729 

This PR tracks taps on the Post Card Footer Link

Taps are tracked as MY_SITE_DASHBOARD_CARD_FOOTER_ACTION_TAPPED with the following properties:
- type = “post”
- sub_type = “create_first”, “create_next”, “drafts”, “scheduled”

`DashboardCardsTracker` was introduced as a helper class to consolidate logic.This class will also be used when implementing Dashboard card item taps.

**Opinion Requested**
 I am considering the same approach for menu/site "items" taps. I'm going back-n-forth on whether I should change the tracker to `MySiteCardsAndItemsTracker` and consolidate the tracking into a single class. Or would it be best to introduce a  `MySiteItemsTracker` tracker for the menu/site items? Thoughts?

**To test:**
Setup
- Launch the app
- Navigate to Me -> App Settings -> Debug Settings
- Ensure that the `MySiteDashboardPhase2FeatureConfig` is to set to on
- Restart the app if needed
- Navigate to Me -> App Settings -> Privacy Settings 
- Toggle the "collect information" switch to on
- Navigate to the My Site tab
------------------------
- Select a site that has no posts
- Tap on the footer link "Create a post" from within the "Create your first post" card
- Navigate to Me -> Help and Support -> Application log
- Note that the following is shown in the log
   I: 🔵 Tracked: my_site_dashboard_card_footer_action_tapped, Properties: {"type":"post","subtype":"create_first"}
-----------------------
- Select a site that has posts, but no draft or scheduled posts
- Tap on the footer link "Create a post" from within the "Create your first post" card
- Navigate to Me -> Help and Support -> Application log
- Note that the following is shown in the log
 I: 🔵 Tracked: my_site_dashboard_card_footer_action_tapped, Properties: {"type":"post","subtype":"create_next"}
-----------------------
- Select a site that has draft posts
- Tap on the footer link "Go to drafts" from within the "Drafts" card
- Navigate to Me -> Help and Support -> Application log
- Note that the following is shown in the log
I: 🔵 Tracked: my_site_dashboard_card_footer_action_tapped, Properties: {"type":"post","subtype":"draft"}
-----------------------
- Select a site that has scheduled posts
- Tap on the footer link "Go to scheduled posts" from within the "Scheduled" card
- Navigate to Me -> Help and Support -> Application log
- Note that the following is shown in the log
I: 🔵 Tracked: my_site_dashboard_card_footer_action_tapped, Properties: {"type":"post","subtype":"scheduled"}

## Regression Notes
1. Potential unintended areas of impact N/A
2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)
DashboardCardsTrackerTest was added to verify event tracking

PR submission checklist:
